sdk/communication/azure-communication-rooms/azure/communication/rooms/_rooms_client.py

```diff
@@ -0,0 +1,300 @@
+        :type room_id: str
+        :returns: None.
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        """
+        self._rooms_service_client.rooms.delete_room(room_id=room_id, **kwargs)
+
+    @distributed_trace
+    def update_room(
+        self,
+        *,
+        room_id: str,
+        valid_from: Optional[datetime] = None,
+        valid_until: Optional[datetime] = None,
+        room_join_policy: Optional[RoomJoinPolicy] = None,
+        participants: Optional[List[RoomParticipant]] = None,
+        **kwargs
+    ) -> CommunicationRoom:
+        """Update a valid room's attributes
+
+        :param room_id: Required. Id of room to be updated
+        :type room_id: str
+        :param valid_from: The timestamp from when the room is open for joining. The timestamp is in
+         RFC3339 format: ``yyyy-MM-ddTHH:mm:ssZ``.
+        :type valid_from: (Optional)datetime
+        :param valid_until: The timestamp from when the room can no longer be joined. The timestamp
+         is in RFC3339 format: ``yyyy-MM-ddTHH:mm:ssZ``.
+        :type valid_until: (Optional)datetime
+        :param room_join_policy: (Optional)The join policy of the room.
+        :type room_join_policy: (Optional)RoomJoinPolicy
+        :param participants: (Optional) Collection of identities invited to the room.
+        :type participants: (Optional)List[RoomParticipant]
+        :returns: Updated room.
+        :rtype: ~azure.communication.rooms.CommunicationRoom
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError
+
+        """
+
+        update_room_request = UpdateRoomRequest(
+            valid_from=valid_from,
+            valid_until=valid_until,
+            room_join_policy=room_join_policy,
+            participants=[p.to_room_participant_internal() for p in participants] if participants else None
+        )
+        update_room_response = self._rooms_service_client.rooms.update_room(
+            room_id=room_id, patch_room_request=update_room_request, **kwargs)
+        return CommunicationRoom.from_room_response(update_room_response)
+
+    @distributed_trace
+    def get_room(
+        self,
+        room_id: str,
+        **kwargs
+    ) -> CommunicationRoom:
+        """Get a valid room
+
+        :param room_id: Required. Id of room to be fetched
+        :type room_id: str
+        :returns: Room with current attributes.
+        :rtype: ~azure.communication.rooms.CommunicationRoom
+        :raises: ~azure.core.exceptions.HttpResponseError
+
+        """
+        get_room_response = self._rooms_service_client.rooms.get_room(room_id=room_id, **kwargs)
+        return CommunicationRoom.from_room_response(get_room_response)
+
+    @distributed_trace
+    def add_participants(
+        self,
+        *,
+        room_id: str,
+        participants: List[RoomParticipant],
+        **kwargs
+    ) -> None:
+        """Add participants to a room
```

Comment:
It would be useful to have a more comprehensive description of the behaviour. For example:
- What happens if I add someone already in the room (i.e. participant duplicates)?
- Is there a maximum number of participants? And what happens when I reach it?